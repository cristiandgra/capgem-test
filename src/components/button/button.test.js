/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import Button from "./button.component";

describe("Button component", () => {
  let wrapper = shallow(<Button />);

  it("should render Button component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Button component with props", () => {
    const mockProps = {
      children: "Button",
      onClick: () => {},
      type: "button",
      className: "btn btn-primary",
      disabled: false,
    };
    wrapper = shallow(<Button {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Button component with props disabled", () => {
    const mockProps = {
      children: "Button",
      onClick: () => {},
      type: "button",
      className: "btn btn-primary",
      disabled: true,
    };
    wrapper = shallow(<Button {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
