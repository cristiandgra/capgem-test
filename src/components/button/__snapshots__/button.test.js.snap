// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render Button component 1`] = `ShallowWrapper {}`;

exports[`Button component should render Button component with props 1`] = `ShallowWrapper {}`;

exports[`Button component should render Button component with props disabled 1`] = `ShallowWrapper {}`;
